src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/controllers/ContestedOrderController.java

```diff
@@ -55,4 +76,100 @@ public ResponseEntity<AboutToStartOrSubmitCallbackResponse> stampFinalOrder(
+    @ApiOperation(value = "check hearing date")
+    @ApiResponses(value = {
+        @ApiResponse(code = 200, message = "Callback was processed successfully or in case of an error message is attached to the case",
+            response = AboutToStartOrSubmitCallbackResponse.class),
+        @ApiResponse(code = 400, message = "Bad Request"),
+        @ApiResponse(code = 500, message = "Internal Server Error")})
+    public ResponseEntity<AboutToStartOrSubmitCallbackResponse> checkHearingDate(
+        @NotNull @RequestBody @ApiParam("CaseData") CallbackRequest callback) {
+
+        CaseDetails caseDetails = callback.getCaseDetails();
+        log.info("Received request to check for manage bundle hearing date for Case ID: {}", caseDetails.getId());
+        validateCaseData(callback);
+
+        final List<String> errors = new ArrayList<>();
+
+        try {
+            validateHearingDate(callback);
+        } catch (InvalidCaseDataException invalidCaseDataException) {
+            errors.add(invalidCaseDataException.getMessage());
+        }
+        return ResponseEntity.ok(AboutToStartOrSubmitCallbackResponse
+            .builder()
+            .data(caseDetails.getData())
+            .errors(errors)
+            .build());
+    }
+
+    @PostMapping(path = "/contested/sortUploadedHearingBundles", consumes = APPLICATION_JSON_VALUE, produces = APPLICATION_JSON_VALUE)
+    @ApiOperation(value = "Documents to be viewed in order of newest first at top of the list")
+    @ApiResponses(value = {
+        @ApiResponse(code = 200, message = "Callback was processed successfully or in case of an error message is attached to the case",
+            response = AboutToStartOrSubmitCallbackResponse.class),
+        @ApiResponse(code = 400, message = "Bad Request"),
+        @ApiResponse(code = 500, message = "Internal Server Error")})
+    public ResponseEntity<AboutToStartOrSubmitCallbackResponse> sortHearingBundles(
+        @NotNull @RequestBody @ApiParam("CaseData") CallbackRequest callback) {
+
+        CaseDetails caseDetails = callback.getCaseDetails();
+        log.info("Received request for doc newest first at top of the list for Case ID: {}", caseDetails.getId());
+        validateCaseData(callback);
+        Map<String, Object> caseData = caseDetails.getData();
+        List<String> errors = new ArrayList<>();
+        List<HearingUploadBundleData> hearingBundleDataList = Optional.ofNullable(caseData.get(HEARING_UPLOAD_BUNDLE_COLLECTION))
+            .map(this::convertToHearingBundleDataList).orElse(Collections.emptyList());
+
+        if (!hearingBundleDataList.isEmpty()) {
+            List<HearingUploadBundleData> updateUploadDateList = hearingBundleDataList.stream()
+                    .map(hd -> HearingUploadBundleData.builder()
+                    .id(hd.getId())
+                    .value(HearingBundle.builder()
+                        .hearingBundleDate(hd.getValue().getHearingBundleDate())
```

Comment:
Perhaps move the builder to its own method as this is quite hard to read

Reply:
done